gnfish/translate_sequences.py

```diff
@@ -1250,60 +47,1269 @@ def read_FASTA_entries(fasta_input_file):
     the next three characters of rnaseq each time next is called on it"""
     nu_seqs = read_FASTA_sequences(filename)
     aa_seqs = []
     for seq in nu_seqs:
         aa_seq = str()
         for n in range(0, len(seq[1]), 3):
-            aa_seq += codons_translation(seq[1][n : n + 3])
+            aa_seq += codons_translation(seq[1][n : n + 3], codes, genetic_code)
         aa_seqs.append([seq[0], aa_seq])
     return aa_seqs
 
 
-def translate_sequences_all_sequences(path, data_type):
+def translate_sequences_all_sequences(
+    path, data_type, pattern, out_exten, codes, genetic_code
+):
     files_lst = get_files(path, pattern)
     for filename in files_lst:
         with open(re.sub("\.f.*", out_exten, filename), "w") as file:
-            for seq in AA_seq_creator(filename):
+            for seq in AA_seq_creator(filename, codes, genetic_code):
                 file.write(">")
                 file.write(seq[0])
                 file.write("\n")
                 file.write(seq[1])
                 file.write("\n")
 
 
-if genetic_code in range(1, 18):
-    if directory is not None:
-        for data_type in data_type_lst:
-            logger.info(f"Running directory and {data_type} argument.")
-            translate_sequences_all_sequences(directory + "/*", data_type)
+def main():
+    # Arguments
+    parser = argparse.ArgumentParser(description="Translates sequences.")
+    parser.add_argument(
+        "--genome",
+        help="extract unique hits from genome data stored at genome folder or specify genome type if using --directory argument.",
+        nargs="?",
+        const="genome",
+        type=str,
+    )
+    parser.add_argument(
+        "--rna",
+        help="extract unique hits from rna data stored at Rna folder or specify genome type if using --directory argument.",
+        nargs="?",
+        const="rna",
+        type=str,
+    )
+    parser.add_argument("--directory", help="path to custom folder", type=str)
+    parser.add_argument(
+        "--genetic_code",
+        help="Default 1 -Standard. Look below for the number of other genetic codes.",
+        nargs="?",
+        type=int,
+        default=1,
+    )
+    parser.add_argument(
+        "--pattern",
+        help='custom pattern to find files for translation. Defaul "_final.fas" and "RAW.fas" as backup.',
+        nargs="?",
+        default="final.fas",
+        const="final.fas",
+        type=str,
+    )
+    parser.add_argument(
+        "--out_exten",
+        help='extension of the output file. Default "_translated.fas".',
+        nargs="?",
+        type=str,
+        default="_translated.fas",
+    )
+    args = parser.parse_args()
+    genome = args.genome
+    rna = args.rna
+    directory = args.directory
+    genetic_code = args.genetic_code
+    pattern = args.pattern
+    out_exten = args.out_exten
+    path = os.getcwd()
+    data_type_lst = [genome, rna]
+
+    codes = {
```

Comment:
It would make sense to have the "codes" in a separate file and then import it for example:
`from gnfish.utils.config import genetic_codes`